### PR TITLE
Allow disabling items in Select components

### PR DESCRIPTION
**Pull request by Lukeaber**
Screenshot 2020-11-27 at 15 18 24

### CODE REVIEW

File: packages/react-components/source/react/internal/action-menu-list/ActionMenuList.js [hunk 1]
```diff
@@ -247,15 +246,21 @@ class ActionMenuList extends Component {
...
               <ActionMenuListItem
                 id={getActionId(id, actionId)}
                 key={actionId}
                 focused={index === focusedIndex}
                 icon={icon}
                 svg={svg}
                 onMouseEnter={() => onMouseEnterItem(index)}
-                onClick={e => executeAction(e, onClick, actionId)}
+                onClick={e =>
+                  disabled ? () => {} : executeAction(e, onClick, actionId)
```
Comment (jilliankeenan) [inline packages/react-components/source/react/internal/action-menu-list/ActionMenuList.js:261]: Would it be better to do something like this?

```suggestion
                  !disabled && executeAction(e, onClick, actionId)
```

**Comment by vine77** [comment]
If we're adding `disabled` to ActionSelect items, should we do the same for Select and ButtonSelect?

**Comment by Lukeaber** [comment]
> If we're adding `disabled` to ActionSelect items, should we do the same for Select and ButtonSelect?
Not sure, estate reporting got a design that required this to be added to the actionSelect but theres nothing for the button select or standard select that ive seen.

**Comment by vine77** [comment]
Do you know how much work it would be to also add `disabled` to the other Select components? There is some benefit in making features symmetrical across components. Not sure if they would share any code. Seems valid to have an item be disabled in any type of dropdown.

**Review by melcherry98 — APPROVED** [review]

**Pull request merged**